File: src/main/resources/largest_digit.rb
puts "Enter a number to find the largest digit:"
number = gets.chomp.to_i

largest = 0

# Find the largest digit using a loop
while number > 0
  digit = number % 10
  if digit > largest
    largest = digit
  end
  number = number / 10
end

# Print the largest digit
puts "The largest digit is: #{largest}"

File: src/main/resources/largest_digit.kt
import java.util.Scanner

fun main() {
  val scanner = Scanner(System.`in`)

  print("Enter a number to find the largest digit: ")
  var number = scanner.nextInt()

  var largest = 0

  // Find the largest digit using a loop
  while (number > 0) {
    val digit = number % 10
    if (digit > largest) {
      largest = digit
    }
    number /= 10
  }

  // Print the largest digit
  println("The largest digit is: " + largest)
}